src/CloudSecrets.UnitTests/ConnectedModeSecretsTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Core.Secrets;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarQube.Client;
+
+namespace SonarLint.VisualStudio.CloudSecrets.UnitTests
+{
+    [TestClass]
+    public class ConnectedModeSecretsTests
+    {
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<ConnectedModeSecrets, IConnectedModeSecrets>(
+                MefTestHelpers.CreateExport<ISonarQubeService>());
+        }
+
+        [TestMethod]
+        public void AreSecretsAvailable_ProjectIsUnbound_NotAvailable()
```

Comment:
Strictly speaking the test names are a bit misleading - it's not checking whether the projects are bound or not, it's checking whether the `SonarQubeClient` is connected. Using something like `ServiceIsConnected/IsNotConnected` would be clearer.